### ISSUE TITLE
[BUG] When using %%timeit to measure execution time, a KeyError is raised

### ISSUE THREAD

**Issue by wjsi** [mars-project/mars#494]
**Describe the bug**
When running a GPU-based graph in local mode, a KeyError is raised.

**To Reproduce**
```python
%%timeit 
import mars.tensor as mt 
from mars.session import new_session 
 
N = 10 ** 9 
data = mt.random.uniform(-1, 1, size=(N, 2), gpu=True) 
inside = (mt.linalg.norm(data, axis=1) < 1).sum() 
pi = 4 * inside / N 
print('pi: %.5f' % pi.execute())
```

**Stack**
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-16-734dcf0f87a7> in <module>
----> 1 get_ipython().run_cell_magic('timeit', '', "\nimport mars.tensor as mt\nfrom mars.session import new_session\n\nN = 10 ** 9\ndata = mt.random.uniform(-1, 1, size=(N, 2), gpu=True)\ninside = (mt.linalg.norm(data, axis=1) < 1).sum()\npi = 4 * inside / N\nprint('pi: %.5f' % pi.execute())\n\n")

~/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py in run_cell_magic(self, magic_name, line, cell)
   2350             with self.builtin_trap:
   2351                 args = (magic_arg_s, cell)
-> 2352                 result = fn(*args, **kwargs)
   2353             return result
   2354 

</home/wenjun.swj/miniconda3/lib/python3.7/site-packages/decorator.py:decorator-gen-60> in timeit(self, line, cell, local_ns)

~/miniconda3/lib/python3.7/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

~/miniconda3/lib/python3.7/site-packages/IPython/core/magics/execution.py in timeit(self, line, cell, local_ns)
   1149                     break
   1150 
-> 1151         all_runs = timer.repeat(repeat, number)
   1152         best = min(all_runs) / number
   1153         worst = max(all_runs) / number

~/miniconda3/lib/python3.7/timeit.py in repeat(self, repeat, number)
    202         r = []
    203         for i in range(repeat):
--> 204             t = self.timeit(number)
    205             r.append(t)
    206         return r

~/miniconda3/lib/python3.7/site-packages/IPython/core/magics/execution.py in timeit(self, number)
    159         gc.disable()
    160         try:
--> 161             timing = self.inner(it, self.timer)
    162         finally:
    163             if gcold:

<magic-timeit> in inner(_it, _timer)

/tmp/pycharm_project_727/mars/tensor/core.py in execute(self, session, **kw)
    375 
    376     def execute(self, session=None, **kw):
--> 377         return self._data.execute(session, **kw)
    378 
    379 

/tmp/pycharm_project_727/mars/core.py in execute(self, session, **kw)
    552         if session is None:
    553             session = Session.default_or_local()
--> 554         return session.run(self, **kw)
    555 
    556     def fetch(self, session=None, **kw):

/tmp/pycharm_project_727/mars/session.py in run(self, *tileables, **kw)
    113         tileables = tuple(mt.tensor(t) if not isinstance(t, (Entity, Base)) else t
    114                           for t in tileables)
--> 115         result = self._sess.run(*tileables, **kw)
    116 
    117         for t in tileables:

/tmp/pycharm_project_727/mars/session.py in run(self, *tileables, **kw)
     54         if 'n_parallel' not in kw:
     55             kw['n_parallel'] = cpu_count()
---> 56         res = self._executor.execute_tileables(tileables, **kw)
     57         return res
     58 

/tmp/pycharm_project_727/mars/utils.py in _wrapped(*args, **kwargs)
    383         try:
    384             _kernel_mode.eager = False
--> 385             return func(*args, **kwargs)
    386         finally:
    387             _kernel_mode.eager = None

/tmp/pycharm_project_727/mars/executor.py in execute_tileables(self, tileables, fetch, n_parallel, n_thread, print_progress, mock, compose)
    551 
    552         self.execute_graph(graph, result_keys, n_parallel=n_parallel or n_thread,
--> 553                            print_progress=print_progress, mock=mock)
    554 
    555         results = self._chunk_result

/tmp/pycharm_project_727/mars/executor.py in execute_graph(self, graph, keys, n_parallel, print_progress, mock, no_intermediate, compose, retval)
    491                                          mock_max_memory=self._mock_max_memory,
    492                                          fetch_keys=fetch_keys, no_intermediate=no_intermediate)
--> 493         res = graph_execution.execute(retval)
    494         self._mock_max_memory = max(self._mock_max_memory, graph_execution._mock_max_memory)
    495         if mock:

/tmp/pycharm_project_727/mars/executor.py in execute(self, retval)
    384         # wait until all the futures completed
    385         for future in executed_futures:
--> 386             future.result()
    387 
    388         if retval:

~/miniconda3/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426 
    427             self._condition.wait(timeout)

~/miniconda3/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/miniconda3/lib/python3.7/concurrent/futures/thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

/tmp/pycharm_project_727/mars/executor.py in _execute_operand(self, op)
    297                         if output.key not in deleted_chunk_keys:
    298                             deleted_chunk_keys.add(output.key)
--> 299                             del results[output.key]
    300 
    301                     # clean the predecessors' results if ref counts equals 0

KeyError: 'd671ef374d7984c041f9c485f36be82e'
```

**Additional context**
Using cupy 6.1.0 (cuda 10.0)

**Comment by qinxuye** [comment]
This issue may have been fixed in #639 . The change below is relative.

https://github.com/mars-project/mars/blob/07ef57cbbd7821d0bb0732a2c40178a714502e33/mars/executor.py#L645

Could you please verify the issue again?

**Comment by qinxuye** [comment]
The problem seems solved, if any error found again, I will reopen it.
